### PR TITLE
Add option to randomly duplicate messages

### PR DESCRIPTION
**Pull request by algattik**
## Purpose

* Upstream systems generating events usually offer at-least-once delivery guarantees. Given that there is no mechanism in IoT Hub / Event Hubs for distributed transactions, it is always possible for the client not to receive the response from the server after sending a message, or crashing before persisting the response, and therefore to retry sending an event. This results in message duplicates. When testing event processing solution, it's important to inject such behavior so as to test if/how the system reacts to it, e.g. by deduplication (e.g. https://github.com/Azure-Samples/streaming-at-scale#duplicate-event-handling). Added an option to randomly duplicate a fraction of messages, e.g. setting `DuplicateEvery` to `1000` will duplicate messages with a probability of 1/1000 for each, i.e. on average will duplicate every 1000th message. Note that messages may be duplicated more than once, i.e. duplicated N times with a probability of `1/(DuplicateEvery)^N`

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Run with `DuplicateEvery` set to `2` and verify that on average, 50% of messages are sent once, 25% are sent twice etc.

### CODE REVIEW

File: src/IotTelemetrySimulator/SenderBase.cs [hunk 1]
```diff
@@ -34,7 +35,10 @@ public async Task SendMessageAsync(RunnerStats stats, CancellationToken cancella
                 {
                     await this.SendAsync(msg, cancellationToken);
                     stats.IncrementMessageSent();
-                    break;
+                    if (this.Config.DuplicateEvery <= 0
+                        || this.random.Next(this.Config.DuplicateEvery) != 0)
+                        break;
+                    attempt = 1;
```
[the remark this answers is not in the document]
Comment (algattik) [reply src/IotTelemetrySimulator/SenderBase.cs:41]: We reset it to give the duplicate message all its retry "chances" again. I.e. if the first message succeeds only at the 3rd of 3 retries, and is duplicated, we want the duplicate to again have the chance of 3 retries.

[the remark this answers is not in the document]
Comment (TechPreacher) [reply src/IotTelemetrySimulator/SenderBase.cs:41]: Ah, ok. Yes, duplicates also need the same retry changes. Agreed.

File: README.md [hunk 2]
```diff
@@ -59,6 +59,7 @@ The amount of devices, their names and telemetry generated can be customized usi
 |Header|telemetry header template (see telemetry template)|
 |PartitionKey|optional [partition key](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features#partitions) template for Event Hubs (see telemetry template)|
 |Variables|telemetry variables (see telemetry template)|
+|DuplicateEveryNEvents|if > 0, send duplicates of the given fraction of messages (default = 0)|
```
[the remark this answers is not in the document]
Comment (algattik) [reply README.md]: Added (also added doc for Event Hubs partition Key, #10)

[the remark this answers is not in the document]
Comment (TechPreacher) [reply README.md]: Perfect, thanks!

**Review by TechPreacher — APPROVED** [review]

**Pull request merged**